incubator/hnc/internal/validators/object.go

```diff
@@ -111,18 +112,37 @@ func (o *Object) handle(ctx context.Context, log logr.Logger, op admissionv1beta
 	// If the object wasn't and isn't inherited, we will check to see if the
 	// source can be created without causing any conflict.
 	if !oldInherited && !newInherited {
+		// check selector format
+		invalid, err := isInvalidSelectorChange(inst, oldInst)
+		// If this is a selector change, and the new selector is not valid, we'll deny this operation
+		if invalid {
+			msg := fmt.Sprintf("Invalid Kubernetes labelSelector: %s", err)
+			return deny(metav1.StatusReasonInvalid, msg)
+		}
+
+		// TODO(@ginnyji): modify hasConflict so that it's aware of selectors
 		if yes, dnses := o.hasConflict(inst); yes {
 			dnsesStr := strings.Join(dnses, "\n  * ")
 			msg := fmt.Sprintf("\nCannot create %q (%s) in namespace %q because it would overwrite objects in the following descendant namespace(s):\n  * %s\nTo fix this, choose a different name for the object, or remove the conflicting objects from the above namespaces.", inst.GetName(), inst.GroupVersionKind(), inst.GetNamespace(), dnsesStr)
 			return deny(metav1.StatusReasonConflict, msg)
 		}
 		return allow("source object")
 	}
+	// This is a propagated object.
+	return o.handleInherited(op, newSource, oldSource, inst, oldInst)
+}
+
+func isInvalidSelectorChange(inst, oldInst *unstructured.Unstructured) (bool, error) {
```

Comment:
Why not just return `nil` if everything's ok, and not a bool? That also makes the caller much more clear - right now, you could imagine someone asking "what if `invalid` is true but `err` is nil?"

```
func validateSelectorChange(inst, oldInst *u.U) error {
  oldStr := ...
  newStr := ...
  if oldStr == newStr {
    return nil
  }
  _, err := selectors.GetSelector(inst)
  return err
}
```

Reply:
done